### PR TITLE
reference.md: clarify the limitation of type alias on an enum

### PR DESCRIPTION
**Pull request by nodakai**
Tentatively define the current behavior as the specification.

Cf. rust-lang/rust#26264, rust-lang/rust#28556, rust-lang/rust#30936

### CODE REVIEW

**Comment by alexcrichton** [comment]
cc @rust-lang/lang 

r? @nikomatsakis

File: src/doc/reference.md [hunk 1]
```diff
@@ -1114,6 +1114,16 @@ type Point = (u8, u8);
...
 ```
 
+Currently a type alias to an enum type cannot be used to qualify the
+constructors:
+
+```
+enum E { A }
+type F = E;
+let _: F = E::A;  // OK
+// let _: F = F::A;  // NG
```
Comment (nikomatsakis) [inline src/doc/reference.md]: "NG"? I don't know what this means, maybe "not good"? Can we just write "Doesn't work" or something like that?

**Comment by nodakai** [comment]
@nikomatsakis My bad, corrected

**Pull request merged**